src/layer/tile/GridLayer.js

```diff
@@ -891,6 +908,27 @@ export const GridLayer = Layer.extend({
 		}
 	},
 
+	_tileUpdate(tile, err, el) {
+		if (err) {
+			this.fire('tileerror', {
+				error: err,
+				tile: el,
+				coords: tile.coords
+			});
+			return;
+		}
+
+		el.classList.add('leaflet-tile-loaded');
+		tile.el.replaceWith(el);
```

Comment:
I see, the element is created behind the curtains and replaced in the DOM once it's ready. Nice technique. :+1: